From 704909b6cd77c6f52a95e5483b6c8ef1399d068f Mon Sep 17 00:00:00 2001
From: Razvan-Daniel Mihai <84674+razvan@users.noreply.github.com>
Date: Fri, 29 May 2026 17:33:00 +0200
Subject: Build custom Stackable version

How it was done:

Replace in files in root dir (do not garble .git):

for d in $(ls -p|grep -v /); do find $d -type f -exec sed -i
's/4\.2\.1/4.2.1-stackable0.0.0-dev/g' {} +; done

Replace in all subdirs:

for d in $(ls -d */); do find $d -type f -exec sed -i
's/4\.2\.1/4.2.1-stackable0.0.0-dev/g' {} +; done
---
 committer-tools/kafka-merge-pr.py                               | 2 +-
 gradle.properties                                               | 2 +-
 streams/quickstart/java/pom.xml                                 | 2 +-
 .../java/src/main/resources/archetype-resources/pom.xml         | 2 +-
 streams/quickstart/pom.xml                                      | 2 +-
 tests/kafkatest/__init__.py                                     | 2 +-
 tests/kafkatest/version.py                                      | 2 +-
 7 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/committer-tools/kafka-merge-pr.py b/committer-tools/kafka-merge-pr.py
index 25026faa63..5416c76085 100755
--- a/committer-tools/kafka-merge-pr.py
+++ b/committer-tools/kafka-merge-pr.py
@@ -70,7 +70,7 @@ TEMP_BRANCH_PREFIX = "PR_TOOL"
 
 DEV_BRANCH_NAME = "trunk"
 
-DEFAULT_FIX_VERSION = os.environ.get("DEFAULT_FIX_VERSION", "4.3.1")
+DEFAULT_FIX_VERSION = os.environ.get("DEFAULT_FIX_VERSION", "4.3.1-stackable0.0.0-dev")
 
 ORIGINAL_HEAD = ""
 
diff --git a/gradle.properties b/gradle.properties
index b7ddc9552a..2f63a60448 100644
--- a/gradle.properties
+++ b/gradle.properties
@@ -14,7 +14,7 @@
 # limitations under the License.
 
 group=org.apache.kafka
-version=4.3.1
+version=4.3.1-stackable0.0.0-dev
 scalaVersion=2.13.18
 # Adding swaggerVersion in gradle.properties to have a single version in place for swagger
 swaggerVersion=2.2.39
diff --git a/streams/quickstart/java/pom.xml b/streams/quickstart/java/pom.xml
index 25253367b8..0ba51d2e89 100644
--- a/streams/quickstart/java/pom.xml
+++ b/streams/quickstart/java/pom.xml
@@ -26,7 +26,7 @@
     <parent>
         <groupId>org.apache.kafka</groupId>
         <artifactId>streams-quickstart</artifactId>
-        <version>4.3.1</version>
+        <version>4.3.1-stackable0.0.0-dev</version>
         <relativePath>..</relativePath>
     </parent>
 
diff --git a/streams/quickstart/java/src/main/resources/archetype-resources/pom.xml b/streams/quickstart/java/src/main/resources/archetype-resources/pom.xml
index 9fed9d1496..fc19271a39 100644
--- a/streams/quickstart/java/src/main/resources/archetype-resources/pom.xml
+++ b/streams/quickstart/java/src/main/resources/archetype-resources/pom.xml
@@ -29,7 +29,7 @@
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <kafka.version>4.3.1</kafka.version>
+        <kafka.version>4.3.1-stackable0.0.0-dev</kafka.version>
         <slf4j.version>2.0.16</slf4j.version>
     </properties>
 
diff --git a/streams/quickstart/pom.xml b/streams/quickstart/pom.xml
index 159ecdac2b..9a5cf5d867 100644
--- a/streams/quickstart/pom.xml
+++ b/streams/quickstart/pom.xml
@@ -22,7 +22,7 @@
     <groupId>org.apache.kafka</groupId>
     <artifactId>streams-quickstart</artifactId>
     <packaging>pom</packaging>
-    <version>4.3.1</version>
+    <version>4.3.1-stackable0.0.0-dev</version>
 
     <name>Kafka Streams :: Quickstart</name>
 
diff --git a/tests/kafkatest/__init__.py b/tests/kafkatest/__init__.py
index da3cc84f26..4e8066ceb7 100644
--- a/tests/kafkatest/__init__.py
+++ b/tests/kafkatest/__init__.py
@@ -22,4 +22,4 @@
 # Instead, in development branches, the version should have a suffix of the form ".devN"
 #
 # For example, when Kafka is at version 1.0.0-SNAPSHOT, this should be something like "1.0.0.dev0"
-__version__ = '4.3.1'
+__version__ = '4.3.1-stackable0.0.0-dev'
diff --git a/tests/kafkatest/version.py b/tests/kafkatest/version.py
index acf715b263..c9aa91a972 100644
--- a/tests/kafkatest/version.py
+++ b/tests/kafkatest/version.py
@@ -128,7 +128,7 @@ def get_version(node=None):
         return DEV_BRANCH
 
 DEV_BRANCH = KafkaVersion("dev")
-DEV_VERSION = KafkaVersion("4.3.1")
+DEV_VERSION = KafkaVersion("4.3.1-stackable0.0.0-dev")
 
 LATEST_STABLE_TRANSACTION_VERSION = 2
 # This should match the LATEST_PRODUCTION version defined in MetadataVersion.java
